webpack/assets/javascripts/react_app/components/Layout/components/TaxonomySwitcher/TaxonomyDropdown.js

```diff
@@ -0,0 +1,121 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { ContextSelectorItem } from '@patternfly/react-core';
+import CustomContextSelector from '../CustomContextSelector';
+import { noop } from '../../../../common/helpers';
+import { translate as __ } from '../../../../common/I18n';
+import './TaxonomyDropdown.scss';
+
+const TaxonomyDropdown = ({
+  taxonomyType,
+  currentTaxonomy,
+  taxonomies,
+  changeTaxonomy,
+  anyTaxonomyText,
+  manageTaxonomyText,
+  anyTaxonomyURL,
+  manageTaxonomyURL,
+  ...props
+}) => {
+  const [searchValue, setSearchValue] = useState('');
+  const [isOpen, setIsOpen] = useState(false);
+  const [filteredItems, setFilteredItems] = useState(taxonomies);
+
+  const onToggle = (event, newIsOpen) => {
+    setIsOpen(newIsOpen);
+  };
+  const onSelect = () => {
+    setIsOpen(!isOpen);
+  };
+  const onSearchInputChange = (value, event) => {
+    setSearchValue(event.target.value);
+    onSearchButtonClick();
+  };
+  const onSearchButtonClick = () => {
+    const filtered =
+      searchValue === ''
+        ? taxonomies
+        : taxonomies.filter(item =>
+            item.title.toLowerCase().includes(searchValue.toLowerCase())
+          );
+    setFilteredItems(filtered || []);
+  };
+
+  const staticGroup = {
+    title: __(taxonomyType),
+    items: [
+      {
+        title: __(anyTaxonomyText),
+        href: anyTaxonomyURL,
+        onClick: () => {
+          changeTaxonomy({ title: anyTaxonomyText });
+        },
+        className: `${taxonomyType.toLowerCase()}s_clear`,
+      },
+      {
+        title: __(manageTaxonomyText),
+        href: manageTaxonomyURL,
+        className: taxonomyType.toLowerCase(),
```

Comment:
I saw `taxonomyType.toLowerCase()` being called on several places,
can you set a `const` for it?

Reply:
Great idea, added a const